### PR TITLE
Fix docs about "callback" filter

### PR DESCRIPTION
**Pull request by phansys**
The CI is failing:
```
docs/reference/filter_field_definition.rst:166:Unexpected indentation.
```
I can't figure out where is the indentation issue. I guess I must produce more tries and failure later.

### CODE REVIEW

File: docs/reference/filter_field_definition.rst [hunk 1]
```diff
@@ -10,29 +10,29 @@ These fields are displayed inside the filter box. They allow you to filter the l
...
-* `Sonata\AdminBundle\Form\Type\Filter\ChoiceType`: display 2 widgets, the operator (yes and no) and the value,
-* `Sonata\AdminBundle\Form\Type\Filter\DefaultType`: display 2 widgets, an hidden operator (can be changed on demand) and the value,
-* `Sonata\AdminBundle\Form\Type\Filter\DateType`: display 2 widgets, the operator ( >, >=, <= , <, =) and the value,
-* `Sonata\AdminBundle\Form\Type\Filter\DateRangeType`: display 3 widgets, the operator (between and not between) and the two values,
-* `Sonata\AdminBundle\Form\Type\Filter\DateTimeType`: display 2 widgets, the operator ( >, >=, <= , <, =) and the value,
-* `Sonata\AdminBundle\Form\Type\Filter\DateTimeRangeType`: display 3 widgets, the operator (between and not between) and the two values,
-
-The `Form Type` configuration is provided by the filter itself.
+* ``Sonata\AdminBundle\Form\Type\Filter\NumberType``: display 2 widgets, the operator ( >, >=, <=, <, =) and the value,
+* ``onata\AdminBundle\Form\Type\Filter\ChoiceType``: display 2 widgets, the operator (yes and no) and the value,
```
Comment (VincentLanglet) [inline docs/reference/filter_field_definition.rst]:
```suggestion
* ``Sonata\AdminBundle\Form\Type\Filter\ChoiceType``: display 2 widgets, the operator (yes and no) and the value,
```

**Comment by phansys** [comment]
I'm blind regarding the indentation errors reported by Sphinx.
Friendly ping @OskarStark.

File: docs/reference/filter_field_definition.rst [hunk 2]
```diff
@@ -6,49 +6,68 @@
...
-The `Form Type` configuration is provided by the filter itself.
+These fields are displayed inside the filter box. They allow you to filter the list of entities by a number of different
+methods.
+
+A filter instance is always linked to a Form Type, there are 7 types available::
+
+* ``Sonata\AdminBundle\Form\Type\Filter\NumberType``:: display 2 widgets, the operator ( >, >=, <=, <, =) and the value;
+* ``Sonata\AdminBundle\Form\Type\Filter\ChoiceType``:: display 2 widgets, the operator (yes and no) and the value;
+* ``Sonata\AdminBundle\Form\Type\Filter\DefaultType``:: display 2 widgets, an hidden operator (can be changed on demand) and
+    the value;
```
Comment (jordisala1991) [inline docs/reference/filter_field_definition.rst]: This line is causing problems

Comment (phansys) [reply docs/reference/filter_field_definition.rst]: Yes, this is the situation after my latest commit, but before, other lines were referenced as offending ones.
I've added many changes now trying to fix the build (replaced 1 backtick with 2 where missing, replaced 1 colon with 2 where missing, etc).

Comment (jordisala1991) [reply docs/reference/filter_field_definition.rst]: I would recomend going back to your original changes and see what is failing together

File: docs/reference/filter_field_definition.rst (hunk 1 above)
Comment (phansys) [reply docs/reference/filter_field_definition.rst]: The minimum changes for what it's worth in this PR are working. Thank you.

File: docs/reference/filter_field_definition.rst (hunk 2 above)
Comment (phansys) [reply docs/reference/filter_field_definition.rst]: The minimum changes for what it's worth in this PR are working. Thank you.

**Review by VincentLanglet — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
thanks @phansys

**Pull request merged**